source/dagger/gameplay/team_game/character_controller.cpp

```diff
@@ -0,0 +1,95 @@
+#include "character_controller.h"
+#include "gameplay/team_game/team_game_main.h"
+
+#include "core/game/transforms.h"
+#include "core/core.h"
+#include "core/engine.h"
+#include "core/input/inputs.h"
+#include "core/graphics/sprite.h"
+#include "core/graphics/animation.h"
+#include "core/graphics/shaders.h"
+#include "core/graphics/window.h"
+
+using namespace team_game;
+
+void TeamGameControllerSystem::OnInitialize(Registry& registry_, Entity entity_)
+{
+    InputReceiver& receiver = registry_.get<InputReceiver>(entity_);
+    for (auto command : { "run", "jump", "down", "heavy", "light", "use" })
+    {
+        receiver.values[command] = 0;
+    }
+}
+
+void TeamGameControllerSystem::SpinUp()
+{
+    Engine::Registry().on_construct<InputReceiver>().connect<&TeamGameControllerSystem::OnInitialize>(this);
+        Engine::Dispatcher().sink<KeyboardEvent>().connect<&TeamGameControllerSystem::OnKeyboardEvent>(this);
+}
+
+void TeamGameControllerSystem::OnKeyboardEvent(KeyboardEvent kEvent_)
+{
+    Engine::Registry().view<ControllerMapping>().each([&](ControllerMapping& ctrl_)
```

Comment:
we have a better way to get an input, check out other team's requests or simply go to platformer example.